src/planner/parabolic/ParabolicTimer.cpp

```diff
@@ -113,6 +115,43 @@ std::unique_ptr<aikido::trajectory::Spline> computeParabolicTiming(
   return outputTrajectory;
 }
 
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline> computeParabolicTiming(
+    const aikido::trajectory::Spline& _inputTrajectory,
+    const Eigen::VectorXd& _maxVelocity,
+    const Eigen::VectorXd& _maxAcceleration)
+{
+  const auto stateSpace = _inputTrajectory.getStateSpace();
+  const auto dimension = stateSpace->getDimension();
+
+  if (static_cast<std::size_t>(_maxVelocity.size()) != dimension)
+    throw std::invalid_argument("Velocity limits have wrong dimension.");
+
+  if (static_cast<std::size_t>(_maxAcceleration.size()) != dimension)
+    throw std::invalid_argument("Acceleration limits have wrong dimension.");
+
+  for (std::size_t i = 0; i < dimension; ++i)
+  {
+    if (_maxVelocity[i] <= 0.)
+      throw std::invalid_argument("Velocity limits must be positive.");
+    if (!std::isfinite(_maxVelocity[i]))
+      throw std::invalid_argument("Velocity limits must be finite.");
+
+    if (_maxAcceleration[i] <= 0.)
+      throw std::invalid_argument("Acceleration limits must be positive.");
```

Comment:
Shouldn't be fine to allow zero velocity/acceleration limits? Could you explain for me why nonnegative limits aren't enough? :sweat_smile:

Reply:
This is following the code of `computeParabolicTiming` for an `Interpolated`.
I think the reason is that if the velocity/acceleration limit of one dimension is zero, it should not be considered in timing a trajectory.